extensions/oplogPopulator/OplogPopulator.js

```diff
@@ -64,10 +63,61 @@ class OplogPopulator {
             const password = encodeURIComponent(authCredentials.password);
             cred = `${username}:${password}@`;
         }
-        let connectionUrl = `mongodb://${cred}${replicaSetHosts}/` +
+        this._mongoUrl = `mongodb://${cred}${replicaSetHosts}/` +
             `?w=${writeConcern}&readPreference=${readPreference}`;
-        if (replicaSet) {
-            connectionUrl += `&replicaSet=${replicaSet}`;
+        this._replicaSet = replicaSet;
+        this._database = database;
+
+        // Promisifying callback functions
+        this._MongoClientConnect = util.promisify(MongoClient.connect);
+    }
+
+    /**
+     * Connects to MongoDB using the MongoClientInterface
+     * and retreives the metastore collection
+     * @returns {Promise|undefined} undefined
+     * @throws {InternalError}
+     */
+    async _setupMongoClient() {
+        try {
+            const client = await this._MongoClientConnect(this._mongoUrl, {
+                 replicaSet: this._replicaSet,
+                 useNewUrlParser: true,
+            });
+            // connect to metadata DB
+            this._mongoClient = client.db(this._database, {
+                ignoreUndefined: true,
+            });
+            // get metastore collection
+            this._metastore = this._mongoClient.collection(constants.bucketMetastore);
+            this._logger.debug('Connected to MongoDB', {
+                method: 'OplogPopulator._setupMongoClient',
+            });
+            return undefined;
+        } catch (err) {
+            this._logger.error('Could not connect to MongoDB', {
+                method: 'OplogPopulator._setupMongoClient',
+                error: err.message,
+            });
+            throw errors.InternalError.customizeDescription(err.message);
+        }
+    }
+
+    /**
+     * Sets default connector configuration
+     * @param {Object} connectorConfig custom connector configuration
+     * @param {string} connectorConfig.name connector name
+     * @returns {Object} connector configuration
+     */
+    _getDefaultConnectorConfiguration(connectorConfig) {
```

Comment:
The mongodb connection code you are referring to i'm assuming is in arsenal's MongoClientInterface ? this part of the code is directly in the constructor so it can't be used